core/GlobalState.cc

```diff
@@ -113,6 +113,79 @@ unique_ptr<GlobalState> GlobalState::makeEmptyGlobalStateForHashing(spdlog::logg
     return rv;
 }
 
+struct MethodBuilder {
+    GlobalState &gs;
+    MethodRef method;
+
+    MethodBuilder(GlobalState &gs, MethodRef m) : gs(gs), method(m) {}
+
+    MethodBuilder &defaultArg(NameRef name) {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isDefault = true;
+        return *this;
+    }
+
+    MethodBuilder &blockArg() {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, Names::blkArg());
+        arg.flags.isBlock = true;
+        return *this;
+    }
```

Comment:
Every method needs a block argument. What do you think about making the `.build()` be explicit, instead of happing on implicit conversion to a `MethodRef`, and moving this block arg stuff to that method?

Reply:
Done.